src/main/java/classes/Duke.java

```diff
@@ -0,0 +1,424 @@
+                    .append("    File 'storage' cannot be found in the folder 'data'.\n")
+                    .append("    A new file 'storage' has been created for you under the folder 'data'\n")
+                    .append("    for storing the tasks!\n")
+                    .append("    ____________________________________________________________\n");
+            Path filePath4 = Paths.get("data/storage.txt");
+            Files.createFile(filePath4);
+            printMessage(pw, sb);
+        } catch (IOException e) {
+            System.out.println("An unexpected error has occurred: " + e.getMessage());
+        } finally {
+            FileWriter fw = new FileWriter(filePath, true);
+            if (fileTasks.size() != 0) {
+                fw.write(textToAdd);
+            } else {
+                for (int i = 0; i < taskList.size(); i++) {
+                    fw.write(taskList.get(i).getTaskInfo() + "\n");
+                }
+            }
+            fw.close();
+        }
+    }
+
+    private static void writeToFile(String oldText, String newText, int oldTextIndex, PrintWriter pw,
+                                    StringBuilder sb, ArrayList<Task> taskList) throws IOException {
+        ArrayList<String> fileTasks = new ArrayList<>();
+        try {
+            fileTasks = getFileContents("data/storage.txt", "data");
+        } catch (FolderNotFoundException e) {
+            Path filePath1 = Paths.get("data/storage.txt");
+            Files.createDirectories(filePath1.getParent());
+            Files.createFile(filePath1);
+            System.out.println(e);
+        } catch (FileNotFoundException e) {
+            sb.append("    ____________________________________________________________\n")
+                    .append("    File 'storage' cannot be found in the folder 'data'.\n")
+                    .append("    A new file 'storage' has been created for you under the folder 'data'\n")
+                    .append("    for storing the tasks!\n")
+                    .append("    ____________________________________________________________\n");
+            Path filePath2 = Paths.get("data/storage.txt");
+            Files.createFile(filePath2);
+            printMessage(pw, sb);
+        } catch (IOException e) {
+            System.out.println("An unexpected error has occurred: " + e.getMessage());
+        } finally {
+            FileWriter fw = new FileWriter("data/storage.txt");
+            if (fileTasks.size() != 0) {
+                for (int i = 0; i < fileTasks.size(); i++) {
+                    if ( (fileTasks.get(i).equals(oldText)) && (i == oldTextIndex) ) {
+                        if (newText.equals("")) {
+                            continue;
+                        }
+                        fw.write(newText + "\n");
+                        continue;
+                    }
+                    fw.write(fileTasks.get(i) + "\n");
```

Comment:
Could this perhaps be simplified?

Reply:
I have yet to update with "A-MoreOOP" section, do stay tune to its release.